MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -2775,17 +2830,17 @@ public IEnumerable<BaseItem> GetThemeVideos()
 
         public IEnumerable<BaseItem> GetExtras()
         {
-            return ThemeVideoIds.Select(LibraryManager.GetItemById).Where(i => i.ExtraType.Equals(Model.Entities.ExtraType.ThemeVideo)).OrderBy(i => i.SortName);
+            return ExtraIds.Select(LibraryManager.GetItemById).Where(i => i != null).OrderBy(i => i.SortName);
         }
 
-        public IEnumerable<BaseItem> GetExtras(ExtraType[] unused)
+        public IEnumerable<BaseItem> GetExtras(ExtraType[] extraTypes)
         {
-            return GetExtras();
+            return ExtraIds.Select(LibraryManager.GetItemById).Where(i => i != null && extraTypes.Contains(i.ExtraType.Value)).OrderBy(i => i.SortName);
         }
 
         public IEnumerable<BaseItem> GetDisplayExtras()
```

Comment:
what's the difference between `GetDisplayExtras()` and `GetExtras()` with no args? are there any more extras that are not listed in `DisplayExtraTypes`?

Reply:
`GetDisplayExtras` retrieves all "specials" for a certain movie. `GetExtras` without parameter will return all extras. In addition to the ones in `DisplayExtraTypes` this would be Trailer, ThemeSong and ThemeVideo
You can find the remaining ones in [MediaBrowser.Model/Entities/ExtraType.cs](https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Model/Entities/ExtraType.cs)